ext_mongodb.php

```diff
@@ -667,7 +667,31 @@ function bsonUnserialize(array $data) : void;
 
 final class Binary implements Type, \Serializable
 {
-	use DenySerialization;
+	static private function checkArray(array $state)
+	{
+		if (
+			!array_key_exists( 'data', $state ) || !is_string( $state['data'] )
+			||
+			!array_key_exists( 'type', $state ) || !is_int( $state['type'] )
+		) {
+			throw new \MongoDB\Driver\Exception\InvalidArgumentException( "MongoDB\BSON\Binary initialization requires \"data\" string and \"type\" integer fields" );
```

Comment:
Must have been a copy and paste from C++ somewhere? I think it's because in C++, I *need* to do this. Just hadn't noticed it. I can fix them.